### PR TITLE
Add general not found page

### LINKED ISSUE #5804: Not found page
> As a user I want to be able to recover from a not found page so that I can navigate to my files again.

## Acceptance Criteria
- A not found page is available
- The not found page contains a link to ownCloud Web

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
We've added a not found page to communicate to users when they've navigated to a page that doesn't exist.

## Related Issue

- Fixes #5804 

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Screenshots (if appropriate):

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

**Review by kulmann — APPROVED** [review]
Love it! 😍  Care to add an e2e test for the page?

File: packages/web-runtime/src/pages/notFound.vue [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<template>
+  <div class="oc-flex oc-flex-center oc-flex-middle oc-flex-column page-not-found">
+    <oc-icon name="emotion-normal" fill-type="line" size="xxlarge" />
+    <h1 class="oc-text-muted">404</h1>
+    <p
+      class="oc-text-xlarge oc-m-rm oc-text-muted"
+      v-text="$gettext('The requested page you are looking for does not exist.')"
```
Comment (kulmann) [inline packages/web-runtime/src/pages/notFound.vue:7]:
```suggestion
      v-text="$gettext('The page you are looking for does not exist.')"
```

File: packages/web-runtime/src/pages/notFound.vue [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    <h1 class="oc-text-muted">404</h1>
+    <p
+      class="oc-text-xlarge oc-m-rm oc-text-muted"
+      v-text="$gettext('The page you are looking for does not exist.')"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { $gettext } from '@ownclouders/web-pkg/src/router/utils'
```
Comment (JammingBen) [inline packages/web-runtime/src/pages/notFound.vue:13]: The import path looks weird, something your IDE did automatically? The import shouldn't be necessary at all, no?

Comment (AlexAndBear) [reply packages/web-runtime/src/pages/notFound.vue:13]: yes over and over again, intelliJ is doing this

File: tests/e2e/cucumber/features/navigation/pageNotFoud.feature [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Feature: Page not found
+  As a user
+  I want to see a not found page
+  So that I know when I've navigated to a page that doesn't exist
+
+  Scenario: not found page
```
[the remark this answers is not in the document]
Comment (kulmann) [reply tests/e2e/cucumber/features/navigation/pageNotFoud.feature:6]: As we found out in a meeting: we can't offer the notFound page in an anonymous context at the moment: user doesn't get initialized and more importantly, apps don't get initialized in an anonymous context. We might change that in the future, but for now we need to accept that the notFound page only works in user- and publicLink-contexts.

**Pull request merged**